### PR TITLE
Setting priorityClassName for system critical pods to system-node-critical

### PR DESCRIPTION
**Pull request by transhapHigsn**
Problem: CoreDNS getting preempted by other pods

Solution: Set priorityClassName to  system-node-critical of traefik, metrics-server, local storage and coredns deployment
Signed-off-by: transhapHigsn <fet.prashantsingh@gmail.com>

#### Proposed Changes ####

Set priorityClassName to `system-node-critical` of traefik, metrics-server, local storage and coredns deployment

#### Types of Changes ####

It is a bug fix for the situation where k3s system critical pods get preempted due to lack of resources.

#### Linked Issues ####

Issues: https://github.com/rancher/k3s/issues/2528

#### Further Comments ####

No comments.

### CODE REVIEW

File: manifests/coredns.yaml [hunk 1]
```diff
@@ -95,6 +95,7 @@ spec:
       labels:
         k8s-app: kube-dns
     spec:
+      priorityClassName: "system-node-critical"
```
Comment (brandond) [inline manifests/coredns.yaml:98]:
```suggestion
      priorityClassName: "system-cluster-critical"
```

File: manifests/traefik.yaml [hunk 2]
```diff
@@ -16,6 +16,7 @@ spec:
     kubernetes:
       ingressEndpoint:
         useDefaultPublishedService: true
+    priorityClassName: "system-node-critical"
```
Comment (brandond) [inline manifests/traefik.yaml:19]:
```suggestion
    priorityClassName: "system-cluster-critical"
```

**Review by brandond — CHANGES_REQUESTED** [review]
This looks great! I suggested a couple changes for components that I think should actually be cluster critical, not node critical.

Also, this isn't documented anywhere, but if you change the manifests you need to follow the instructions in BUILDING.md to to prepare your environment to build k3s, and then run `go generate` and check in the updated generated bindata files. You don't actually need to run a full build; just prepare the local dev environment.

**Comment by transhapHigsn** [comment]
@brandond this command `mkdir -p build/data && ./scripts/download && go generate` from BUILDING.md has made some changes in `pkg/deploy/zz_generated_bindata.go`file, are you referring to pushing those changes in the pr?

**Comment by transhapHigsn** [comment]
@brandond I have messed up commits in this PR, will be create new from scratch. Can you decline this one?

**Comment by transhapHigsn** [comment]
I have messed up commits in this PR, will create new from scratch.

**Comment by transhapHigsn** [comment]
New pull request: https://github.com/rancher/k3s/pull/2558

**Pull request closed without merging**